binderhub/builder.py

```diff
@@ -542,31 +596,29 @@ def _check_result(future):
                         BUILD_COUNT.labels(
                             status="failure", **self.repo_metric_labels
                         ).inc()
-
                 await self.emit(event)
 
-        # Launch after building an image
+        if build_only_outcome:
+            return
+
         if not failed:
-            BUILD_TIME.labels(status="success").observe(
-                time.perf_counter() - build_starttime
-            )
-            BUILD_COUNT.labels(status="success", **self.repo_metric_labels).inc()
+            # Launch after building an image
             with LAUNCHES_INPROGRESS.track_inprogress():
                 await self.launch(provider)
-            self.event_log.emit(
-                "binderhub.jupyter.org/launch",
-                5,
-                {
-                    "provider": provider.name,
-                    "spec": spec,
-                    "ref": ref,
-                    "status": "success",
-                    "build_token": self._have_build_token,
-                    "origin": self.settings["normalized_origin"]
-                    if self.settings["normalized_origin"]
-                    else self.request.host,
-                },
-            )
+                self.event_log.emit(
+                    "binderhub.jupyter.org/launch",
+                    5,
+                    {
+                        "provider": provider.name,
+                        "spec": spec,
+                        "ref": ref,
+                        "status": "success",
+                        "build_token": self._have_build_token,
+                        "origin": self.settings["normalized_origin"]
+                        if self.settings["normalized_origin"]
+                        else self.request.host,
+                    },
+                )
```

Comment:
The indentation here seems unrelated to the PR. The indentation change could make sense, but I'm not sure. It would influence the time of the prometheus metric a bit, decrementing LAUNCHES_INPROGRESS after event_log.emit() has been called also.

Since the BUILD_TIME tracking is resolved before emitting the event, I suggest we revert this indentation for consistency and to make this PR scoped tighter

```
            with LAUNCHES_INPROGRESS.track_inprogress():
                await self.launch(provider)
```

Reply:
Oh, yes. I think the inndentation is like this by mistake. Thanks for spotting it